rpcserver.go

```diff
@@ -2459,6 +2459,33 @@ func (r *rpcServer) ListPeers(ctx context.Context,
 			Features:  features,
 		}
 
+		var peerErrors []interface{}
+
+		// If we only want the most recent error, get the most recent
+		// error from the buffer and add it to our list of errors if
+		// it is non-nil. If we want all the stored errors, simply
+		// add the full list to our set of errors.
+		if in.LatestError {
+			latestErr := serverPeer.errorBuffer.Latest()
+			if latestErr != nil {
+				peerErrors = []interface{}{latestErr}
+			}
+		} else {
+			peerErrors = serverPeer.errorBuffer.List()
+		}
+
+		// Add the relevant peer errors to our response.
+		for _, error := range peerErrors {
+			tsError := error.(*timestampedError)
```

Comment:
To avoid a panic _if_ there ever is something other in there, we could just skip the item if the cast is not ok.

Reply:
If we do make a change which adds something else, I'm pretty confident that the itests will panic. Should be picked up developer end when testing changes anyway, so I'd rather fail loudly?

Reply:
Yes, failing loudly also sounds good to me. It _should_ never happen but a nice error is always nicer than just a panic.